Code/Tools/AssetProcessor/native/utilities/Builder.cpp

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <QCoreApplication>
+#include <QElapsedTimer>
+#include <AzCore/Settings/CommandLine.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/std/parallel/binary_semaphore.h>
+#include <AzCore/Utils/Utils.h>
+#include <utilities/Builder.h>
+#include <utilities/AssetBuilderInfo.h>
+
+namespace AssetProcessor
+{
+    //! Amount of time in milliseconds to wait between checking the status of the AssetBuilder process and pumping the stdout/err pipes
+    constexpr int s_MaximumSleepTimeMS = 10;
+
+    //! Time in milliseconds to wait after each message pump cycle
+    constexpr int s_IdleBuilderPumpingDelayMS = 100;
+
+    constexpr int s_MillisecondsInASecond = 1000;
+
+    constexpr const char* s_buildersFolderName = "Builders";
```

Comment:
any reason why you need to spaces these out so much?

Reply:
Not really, the first two were spaced out to leave room for the comments since it looks kinda weird when they're jammed together.  The others could probably move up I guess.